Assignment0/README.md

```diff
@@ -0,0 +1,24 @@
+# Matrix-Vector Product Implementation
+
+This project implements a C++ function to compute the product of a dense matrix with a matrix or a vector. It includes separate files for the implementation, a test suite, and a build system using `make`.
+
+---
+
+## File Structure
+
+- **header.hpp**: Header file with the function declarations.
```

Comment:
`.hpp` already marks the file as a header. The name of the file should be meaningful.